src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseIntegrationBambooBitbucketJiraTest.java

```diff
@@ -1060,4 +1058,38 @@ void testGetTemplateFilesWithContentShouldRedirect() throws Exception {
             return localRepository;
         });
     }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testGetParticipationFilesWithContentAtCommitShouldRedirect() throws Exception {
+        programmingExerciseIntegrationTestService.testRedirectGetParticipationRepositoryFilesWithContentAtCommit((exercise, files) -> {
+            LocalRepository localRepository = new LocalRepository("main");
+            var studentLogin = TEST_PREFIX + "student1";
+            try {
+                return hestiaUtilTestService.setupSubmission(files, exercise, localRepository, studentLogin);
+            }
+            catch (Exception e) {
+                fail("Test setup failed");
+            }
```

Comment:
Won't the exception make the test fail automatically?

Reply:
Not if you catch it. And we have to catch it because it's part of a lambda expression so we cannot just pass it on to the calling method.

Reply:
I see 👍